test/snippet/alphabet/composition/union_composition.cpp

```diff
@@ -0,0 +1,143 @@
+#include <seqan3/alphabet/composition/union_composition.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/nucleotide/rna4.hpp>
+#include <seqan3/alphabet/gap/gap.hpp>
+#include <gtest/gtest.h>
+
+using namespace seqan3;
+int main()
+{
+
+{
+//! [variant]
+union_composition<dna4, gap> my_letter{};
+union_composition<dna4, gap> converted_letter{dna4::C};
+// doesn't work:
+// union_composition<dna4, gap> my_letter{'A'};
+union_composition<dna4, gap>{}.assign_char('C'); // <- this does!
+union_composition<dna4, gap>{}.assign_char('-'); // gap character
+union_composition<dna4, gap>{}.assign_char('K'); // unknown characters map to the default/unknown
+                                           // character of the first alphabet type (i.e. A of dna4)
+if (my_letter.to_char() == 'A')
+std::cout << "yeah\n"; // "yeah";
+//! [variant]
+}
+
+{
+//! [construct base]
+using alphabet_t = union_composition<dna4, dna5, gap>;
+
+constexpr alphabet_t letter0{dna4::A};
+constexpr alphabet_t letter1 = dna4::C;
+constexpr alphabet_t letter2 = {dna4::G};
+constexpr alphabet_t letter3 = static_cast<alphabet_t>(dna4::T);
+
```

Comment:
Please add an construction from `dna5` and from `gap::GAP` or replace them for some of the dna4 above. This is a union composition and it should be clear that you can construct from any type that is part of the composition.